08openmp/index.md

```diff
@@ -0,0 +1,307 @@
+---
+title: "Week 8: Parallel Programming with OpenMP"
+---
+
+## Overview
+
+This week we'll be focusing on speeding up our code by taking advantage of **parallelism**, i.e. splitting up problem and giving different pieces to different processors. In these notes we'll introduce the following concepts:
+
+- What is parallel programming?
+  - Why do we need parallel programming?
+  - What are shared- and distributed-memory parallelism?
+  - What are weak and strong scaling?
+- An introduction to OpenMP
+  - Parallelising loops
+  - Reductions
+  - Data sharing
+  - Thread control
+  - Schedules
+  - Parallelising multi-dimensional loops
+
+## What is parallel programming?
+
+So far in this course we've been writing fundamentally **sequential** programs, where the code runs from start to finish, each step one after another. Our programs don't try and do multiple operations at once. However, modern **multi-core** CPUs are perfectly capable of performing multiple operations in **parallel**. Our sequential programs are like single queues at a supermarket checkout, capable of processing only one person's items at a time. Open more lanes and we can process a lot more people! If we can split up our problems appropriately and run each piece in parallel, we should be able to speed up our code immensely.
```

Comment:
```suggestion
So far in this course we've been writing fundamentally **sequential** programs, where the code runs from start to finish, each step one after another. Our programs don't try to do multiple operations at once. However, modern **multi-core** CPUs are perfectly capable of performing multiple operations in **parallel**. Our sequential programs are like single queues at a supermarket checkout, capable of processing only one person's items at a time. Open more lanes and we can process a lot more people! If we can split up our problems appropriately and run each piece in parallel, we should be able to speed up our code immensely.
```
This is slightly complicated a bit by the fact that we introduced SIMD registers and operations last week, so there has been a tiny bit of parallelism. You can mention this for completeness but I don't think it's very confusing to gloss over either.